### ISSUE TITLE
Open file descriptor on read stream

### ISSUE THREAD

**Issue by shaunwarman** [alanhoff/node-tar.gz#23]
Seeing `ENOENT, open 'path/to/file/that/was/streamed'`

Is there a way to auto close the read stream when trying to remove the directory where the tar was created?

**Comment by shaunwarman** [comment]
To elaborate:
```
var read = targz().createReadStream(path);
var write = fs.createWriteStream(path);
read.pipe(write);
```

Creates an open file descriptor:
```
node      65975 swarman   22w     REG                1,4          0 221989999 /private/tmp/app_name/export/MyAwesomeFilev2.1.tar.gz
```

Now when I try to cleanup this directory: `/tmp/app_name/export` it spits out an `ENOENT, open ...` error

**Comment by alanhoff** [comment]
Must investigate why it's not closing the file descriptor

**Comment by alanhoff** [comment]
Fixed by #28
